modules/linear_boltzmann_solvers/executors/pi_keigen_smm.cc

```diff
@@ -867,16 +867,15 @@ PowerIterationKEigenSMM::ComputeAuxiliaryUnitCellMatrices()
     const auto num_cell_nodes = cell_mapping.NumNodes();
     const auto qp_data = cell_mapping.MakeVolumetricFiniteElementData();
 
-    std::vector<std::vector<std::vector<std::vector<double>>>> K;
-    K.resize(num_cell_nodes, std::vector<std::vector<std::vector<double>>>(num_cell_nodes));
+    NDArray<double> K(std::vector<size_t>({num_cell_nodes, num_cell_nodes, dimension_, dimension_}),
+                      0.);
```

Comment:
For whatever reason, this zero by itself on the next line is very unsatisfying... No need to change it unless it bugs you too